crates/net/network/src/transactions.rs

```diff
@@ -1016,6 +1051,84 @@ struct Peer {
     client_version: Arc<String>,
 }
 
+// === TransactionFetcher ===
+/// * active requests - especially inflight hashes, so that whenever we receive new missing hashes
+///   we can check first if they're already being requested.
+/// * missing hashes and peers that send us these - so we can possibly re-request them
+
+#[derive(Debug)]
+struct TransactionFetcher {
+    /// All currently active requests for pooled transactions.
+    inflight_requests: FuturesUnordered<GetPooledTxRequestFut>,
+
```

Comment:
what we need here is a set that tracks __all__ hashes that are currently being fetched, so we don't need to loop over all requests in `in_flight`

Reply:
I added this as `inflight_hash_to_fallback_peers: HashMap<TxHash, Vec<PeerId>>`, not sure thats the optimal structure, very interested in your feedback